src/main/flight/ground_frame.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * This file is part of Betaflight.
+ *
+ * Betaflight is free software. You can redistribute this software
+ * and/or modify this software under the terms of the GNU General
+ * Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * Betaflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <math.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+#include "platform.h"
+
+#include "build/debug.h"
+#include "fc/core.h"
+#include "fc/rc_modes.h"
+#include "fc/runtime_config.h"
+
+#include "rx/rc_stats.h"
+
+#include "sensors/acceleration.h"
+#include "flight/imu.h"
+
+float groundFrameAcc[3];
+float groundFrameAccMagnitude;
+
+void updateGroundFrame(void)
+{
+    const float gravity = acc.dev.acc_1G;
+    const float (*rMat)[3] = getCraftRotationMaxtrix();
+    float transposedMat[3][3];
+    transposeMatrix3x3(rMat, transposedMat);
+
+    // Rotate the gravity vector [0, 0, g] from the earth frame to the body frame
+    const float gravity_vector_body[3] = {
```

Comment:
Inlined matrix*vector multiplication function would be optimized, resulting in easier to read code with similar performance

Reply:
i guess you are right, don't even need to rotate gravity, can just substract it in ground frame